packages/ddex/entrypoint/src/components/DistributorCard.tsx

```diff
@@ -1,19 +1,54 @@
-import { Flex, Text } from "@audius/harmony"
+import {
+  useQuery,
+} from '@tanstack/react-query'
+import { sdk } from '@audius/sdk'
+import { Flex, IconEmbed, Text } from "@audius/harmony"
+import { useSdk } from '../contexts/AudiusSdkProvider'
+import { useCallback } from 'react'
+import { Status } from '../contexts/types'
+
+const env = import.meta.env.VITE_ENVIRONMENT as 'dev' | 'stage' | 'prod'
 
 type DistributorCardProps = {
-  name: string
-  imageUrl: string
-  onClick: () => void
+  appKey: string
+  url: string
 }
 
 export const DistributorCard = ({
-  name,
-  imageUrl,
-  onClick
+  appKey,
+  url
 }: DistributorCardProps) => {
+  const {sdk: audiusSdk, status} = useSdk()
+
+  const { data } = useQuery({
+    queryKey: ['todos'],
+    queryFn: () =>
+      audiusSdk
+        ?.developerApps.getDeveloperApp({ address: appKey })
+        .then(res => res.data),
+    enabled: status !== Status.IDLE && status !== Status.LOADING
+  })
+
+  const handleClick = useCallback(() => {
+    // Initialize an sdk for the distributor and use that to auth the user.
+    const distroSdk = sdk({
+      apiKey: appKey
+    })
+    distroSdk.oauth!.init({
+      successCallback: () => undefined,
+      env: env === 'prod' ? 'production' : 'staging'
+    })
+    distroSdk.oauth!.login({
+      scope: 'write',
+      redirectUri: `${url}/auth/redirect?redirect_uri=${window.location.origin}`,
```

Comment:
probably should `encodeURIComponent(window.location.origin)`

Reply:
origin doesn't need to be URI encoded!

Reply:
but I will do anyway